test/expect.js

```diff
@@ -602,6 +602,16 @@ describe('expect', function () {
     expect({ foo: 1, bar: 2 }).to.not.have.all.keys(['baz', 'foo']);
     expect({ foo: 1, bar: 2 }).to.not.contain.all.keys(['baz', 'foo']);
 
+    // issues/359
+    (function fix359() {
+      var expected = [ 'b', 'a' ];
+      var backup = expected.slice(0);
+      var obj = { "b": 1, "a" :1 };
+      expect(expected).deep.equal(backup);
+      expect(obj).keys(expected);
+      expect(expected).deep.equal(backup);
+    })()
+
```

Comment:
Stylistically I think these can just be inlined - I dont think there is a need for a closure. If you want to you could create a new `it()` test instead.